biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -824,7 +837,10 @@ public void run() {
 		if (!workingdir.isDirectory())
 			throw new IllegalArgumentException("Cannot create a working dir: " + workingdir);
 
-		p.setProperty(Constants.FRAMEWORK_STORAGE, workingdir.getAbsolutePath());
+		if (System.getProperty(Constants.FRAMEWORK_STORAGE) == null)
+			p.setProperty(Constants.FRAMEWORK_STORAGE, workingdir.getAbsolutePath());
+		else
+			p.setProperty(Constants.FRAMEWORK_STORAGE, System.getProperty(Constants.FRAMEWORK_STORAGE));
```

Comment:
No, I want it to only make a difference when people use -D in the launching for backward compatibility. And the storagedir gets processed. I think this is fine.